package/MDAnalysis/topology/guessers.py

```diff
@@ -203,7 +200,8 @@ def guess_bonds(atoms, coords, **kwargs):
 
     lower_bound = kwargs.get('lower_bound', 0.1)
 
-    box = kwargs.get('box', None)
+    if box:
```

Comment:
The  `is not None` check is enough. That will reflect the old behavior and the numpy asarray call later ensures us the we have an array.